gapic-generator/lib/gapic/schema/request_param_parser.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gapic/schema/request_parameter"
+require "gapic/schema/parameter_schema"
+
+module Gapic
+  module Schema
+    # Contains logic for parsing protoc request parameters
+    # from the plugin_opt command line
+    module RequestParamParser
+      class << self
+        # Unescapes a symbol from the string
+        # e.g. "a\.b", '.' => "a.b"
+        # @param string [String] String to unescape
+        # @return [String] Unescaped string
+        def unescape string
+          string ? string.gsub(/\\./) { |escaped| escaped[1] } : string
```

Comment:
Do you expect string to be nil sometimes? Is that why we're testing string for truthiness? (If so, include `nil` in the YARD type annotation.)

Reply:
that seems like a holdover from some point in development where nils were a possibility

Reply:
fixed